### ISSUE TITLE
How to use evaluation_metrics.py file

### ISSUE THREAD

**Issue by ABCBeacuse** [alexandor91/se3-equi-graph-registration#7]
Hello, when I was using the evaluation_metrics.py file, I found that I need the gt_file_list, pred_file_list, gt_dir, pred_dir, and save_dir files. How should I create these files needed for calculation based on the trained pth file?

**Comment by ABCBeacuse** [comment]
I didn't find the evaluation.py file in the tools folder (according to the ReadMe file)

**Comment by alexandor91** [comment]
Hi, there, sorry for some sync-up issue  in readme, in train_eval_egnn,py you can use test mode to export the pair input points, features and prediction results into npy file, then under tools folder there is a evaluation_metrics file which u can use to evaluate pairwise scan error for metric evaluation, some common metric are included in the script.

**Comment by ABCBeacuse** [comment]
Hello, thank you very much for your reply. I have a problem with an error when I set mode to test in the train_eval_egnn.py file.
![Uploading QQ20250106-155032.png…]()

**Comment by ABCBeacuse** [comment]
Hello, thank you very much for your reply. But I have a problem with an error when I set mode to test in the train_eval_egnn.py file.
![QQ20250106-155032](https://github.com/user-attachments/assets/a3cb0e52-2dd2-4ba7-881a-0dca77876d7d)

**Comment by ABCBeacuse** [comment]
And then when I set it to val, nothing happens，Maybe I did something wrong。
![QQ20250106-155906](https://github.com/user-attachments/assets/68b3ce85-15e7-4799-90bf-09ac5e082963)

**Comment by alexandor91** [comment]
It should be set to test, I will update the readme recently with more details about how to do evaluation, say tuned please.

**Comment by ABCBeacuse** [comment]
OK. Thank you very much for your reply.

**Comment by ABCBeacuse** [comment]
Hello, could you please improve it recently? Thanks a million!

**Comment by alexandor91** [comment]
Hi there, the main branch is updated now with a separate script "eval_egnn_metrics.py" you can directly use it to load your best trained checkpoint for the on-the-fly evaluation of model, the mean metric results are exported into a txt file.

**Comment by ABCBeacuse** [comment]
Thank you very much, And I'm really sorry for bothering you so often.
There is a new problem.... I used your latest script with the calculation parameters. I loaded a pth file generated by training with your model and also tried pth files from other epochs. I found that the magnitude of the calculated parameters might be a bit off. I'm not sure if it's a problem with my machine. 
![QQ20250115-163928](https://github.com/user-attachments/assets/26c5c311-d4a1-4752-9388-77df9b6cd095)

**Comment by alexandor91** [comment]
May I know your final training loss level? As usually this should indicate that model is not converging, usually 0.1 level rot loss should be good for evaluation. Also if you are using batch size 1 for training now, getting into training local minimum may sometimes happen, thus we suggest to try our new script with batch size support later, as we plan to release it around spring festival. The script with batch enabled now is existing under "debug" branch, we will merge it into main branch after fully porting work. Also as the train and test data will be updated due to the old data link expiry, you may also need to redownload the processed data from the new link along with the new train batch script then.

**Comment by ABCBeacuse** [comment]
I noticed that my final loss is still around level 1, which might be due to the issue you mentioned where a batch size of 1 prevents the model from converging properly. Thank you so much for your detailed and patient explanation. I am eagerly looking forward to your updated script code. Wishing you a Happy New Year!

**Comment by ABCBeacuse** [comment]
Hello, may I ask when the latest version is expected to be released?

**Comment by alexandor91** [comment]
Hi, the code with batch size support is already released under debug branch "train_egnn_with_batch.py", including pre-trained weight and evaluation script "eval_egnn_metrics.py" for 3D Match. For data, we have some problem to to OneDrive expiry problem, so plan to provide data processing script, so use can extract feature locally. otherwise the data is hard to maintain public, I will rebase the debug branch later into main, and replace the data link in readme this week.
